src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/PackageBuilder.cs

```diff
@@ -366,6 +385,58 @@ public void Save(Stream stream)
             }
         }
 
+        private static byte[] ReadAllBytes(Stream stream)
+        {
+            using (var ms = new MemoryStream())
+            {
+                stream.CopyTo(ms);
+                return ms.ToArray();
+            }
+        }
+
+        private string CalcPsmdcpName()
+        {
+            if (_deterministic)
+            {
+                using (var hashFunc = new Sha512HashFunction())
+                {
+                    foreach (var file in Files)
+                    {
+                        var data = ReadAllBytes(file.GetStream());
+                        hashFunc.Update(data, 0, data.Length);
+                    }
+                    return EncodeHexString(hashFunc.GetHashBytes()).Substring(0, 32);
+                }
+            }
+            else
+            {
+                return Guid.NewGuid().ToString("N");
+            }
+        }
+
+        private static readonly char[] HexValues = { '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'A', 'B', 'C', 'D', 'E', 'F' };
+        // Reference https://github.com/dotnet/corefx/blob/2c2e4a599889652ec579a870054b0f8915ea70fd/src/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/Utils.cs#L736
```

Comment:
Is there a reason we're using this implementation instead of something like [`BitConverter.ToString(sArray).Replace("-", "")`](https://docs.microsoft.com/en-us/dotnet/api/system.bitconverter?view=netcore-2.2)?

Reply:
Perf, because this is quite a bit faster, but I think I might be over-optimizing it, consider this will operate on hashes.
I can change it something simpler like that.